parser/lexer.go

```diff
@@ -204,6 +204,15 @@ func (s *Scanner) scan() (tok int, pos Pos, lit string) {
 		ch0 = s.skipWhitespace()
 	}
 	pos = s.r.pos()
+
+	if ch0 == '\\' {
+		// support \N as shortcut for NULL
+		if s.r.charByIndex(pos.Offset+1) == 'N' {
```

Comment:
I think you can simply add `initTokenString("\N", null)` at https://github.com/pingcap/tidb/blob/master/parser/misc.go#L122 .
@tiancaiamao